### PR TITLE
Include /etc/map url mappings in archetype

### PR DESCRIPTION
**Pull request by hearora**
This PR includes changes done for multiple RTCs.

If the user enters domain on archetype execution, configs specific for publish environment will be generated under /etc/map.publish and the corresponding config update will be done in jcrresourceresolver config to point to map.bulish directory for publish envs.

secondly by default 2 templates will be generated(during archetype execution) homepage and landing page with a predefined structure and 2 experience fragments for header footer related to the custom project will also be generated during archetype execution and associated with the pages created using these templates

In addition to this , during archetype execution the user will be prompted for creating runmode configs based on the names of environments entered by the user.

The content skeleton has also been updated. By default for the mid market blueprint projects, new content heirarchy has been introduced. /content/<content_folder_name>/us/en

work is being continued on the content skeleton, based on user input the appropriate content skeleton will be generated.

updated the content page type component to render only a responsive grid instead of OOTB comps included.

A standard data layer js has also been added.

## Related Issue
https://github.com/adobe/aem-project-archetype/issues/163
https://github.com/adobe/aem-project-archetype/issues/179
https://github.com/adobe/aem-project-archetype/issues/174

## Motivation and Context

These updates are done for mid-market blueprint projects

## How Has This Been Tested?

tested execution on local by updating the artifact id, building it and then generating the project by executing archetype using updated artifact id.
a sample project has been generated and is going to be demonstrated next week.

- [x ] New feature (non-breaking change which adds functionality)
- [x ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).

### CODE REVIEW

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/structure/page/customheaderlibs.html [hunk 1]
```diff
@@ -18,3 +18,20 @@
 
 <!--/* Include Context Hub */-->
 <sly data-sly-resource="${symbol_dollar}{'contexthub' @ resourceType='granite/contexthub/components/contexthub'}"/>
+<!--/* Data layer JavaScript object: */-->
+<script  data-sly-use.data="/apps/${appsFolderName}/components/structure/page/digitaldata.js"
```
Comment (vladbailescu) [inline src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/structure/page/customheaderlibs.html]: -1 for using JS-use API. All information is already available in the context.

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/structure/page/customheaderlibs.html [hunk 2]
```diff
@@ -18,3 +18,20 @@
 
 <!--/* Include Context Hub */-->
 <sly data-sly-resource="${symbol_dollar}{'contexthub' @ resourceType='granite/contexthub/components/contexthub'}"/>
+<!--/* Data layer JavaScript object: */-->
```
Comment (vladbailescu) [inline src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/structure/page/customheaderlibs.html]: I would rather we add the datalayer to the Core Components instead, so everyone can benefit from it.

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/structure/page/customheaderlibs.html (hunk 1 above)
Comment (hearora) [reply src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/structure/page/customheaderlibs.html]: Resolved.

**Comment by hearora** [comment]
> Please break down the PR into separate pieces for each new feature/improvement, for easier review!

Creating seperate PRs.

**Comment by hearora** [comment]
> > Please break down the PR into separate pieces for each new feature/improvement, for easier review!
> 
> Creating seperate PRs.
https://github.com/adobe/aem-project-archetype/pull/192
https://github.com/adobe/aem-project-archetype/pull/191
https://github.com/adobe/aem-project-archetype/pull/190

 Kindly close this PR without merge.

**Pull request closed without merging**